heudiconv/convert.py

```diff
@@ -628,3 +627,29 @@ def save_converted_files(res, item_dicoms, bids_options, outtype, prefix, outnam
             except TypeError as exc:  ##catch lists
                 raise TypeError("Multiple BIDS sidecars detected.")
     return bids_outfiles
+
+
+def  add_taskname_to_infofile(infofile):
+    """Add the "TaskName" field to json files corresponding to func images.
+
+    Parameters
+    ----------
+    infofile : filename of json file
+
+    Returns
+    -------
+    """
+
+    import re
+
+    meta_info = load_json(infofile)
+
+    try:
+        meta_info['TaskName'] = (re.search('(?<=_task-)\w+',
+                                           op.basename(infofile))
+                                 .group(0).split('_')[0])
+    except AttributeError:
+        pass
```

Comment:
why to pass and not just return while may be logging that failed to find any task field?

Reply:
You are right, it may just return.
I don't know if it should log that it failed to find a task field because a.f.a.i.k. this function will be called for any multi-echo sequence, and not all of them are `func`.